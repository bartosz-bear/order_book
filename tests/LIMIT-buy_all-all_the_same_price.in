{"type": "Limit", "order": {"direction": "Buy", "id": 1, "price": 14, "quantity": 20}}
{"type": "Limit", "order": {"direction": "Buy", "id": 2, "price": 14, "quantity": 20}}
{"type": "Limit", "order": {"direction": "Buy", "id": 3, "price": 14, "quantity": 20}}
{"type": "Limit", "order": {"direction": "Sell", "id": 4, "price": 14, "quantity": 80}}