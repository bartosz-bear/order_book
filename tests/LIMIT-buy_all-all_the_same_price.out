{"sellOrders": [], "buyOrders": [{"id": 1, "price": 14, "quantity": 20}]}
{"sellOrders": [], "buyOrders": [{"id": 1, "price": 14, "quantity": 20}, {"id": 2, "price": 14, "quantity": 20}]}
{"sellOrders": [], "buyOrders": [{"id": 1, "price": 14, "quantity": 20}, {"id": 2, "price": 14, "quantity": 20}, {"id": 3, "price": 14, "quantity": 20}]}
{"sellOrders": [{"id": 4, "price": 14, "quantity": 20}], "buyOrders": []}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 20, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 20, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 20, "price": 14}